### DISCUSSION TITLE
Struggling to get equivalent of  `find . -name "some_file_pattern" | xargs grep "some_search_text"

### DISCUSSION THREAD

**Discussion by sandwoodK** [nushell/nushell#6693] [General]
One of my usefull shell command is 
`find . -name "some_file_pattern" | xargs grep "search_text"`

I am looking for equivalent in nushell. Ideally I would like to get a table , first column being match file name, second column being a table of found lines matching the search text.

So far I have 
`〉(ls "**/some_file_pattern").name | where { |f| open $f | str contains "search_text"}  | each { |f| open $f | lines | find --regex "search"}`

Not still that, i lose the filename. And search text is repeated in the column. 
Any help appreciated.

**Comment by fdncred** [comment]
I don't use the os's `find` or `xargs`. If you could cook up a prototype from the root of the nushell repo as an example I might be able to help.

**Reply by sandwoodK** [reply]
Some example. 
Find from the nushell root all markdown files which contains "plugin", and display those lines
`find . -name "*.md" | xargs grep plugin`

**Reply by fdncred** [reply]
thanks. here's a start with a tabular output (using ripgrep instead of grep)
```
ls **/*.md | each {|f| rg plugin $f.name --json | from json -o | where type == match} | flatten | get data | flatten
```
but you can also just do this which is nearly the equivalent which shows that in nushell we can use `each` in place of `xargs`
```
ls **/*.md | each {|f| rg plugin $f.name}
```
